### DISCUSSION TITLE
How Can I whitelist/add IP address to ignore list via action.d temporary basis?

### DISCUSSION THREAD

**Discussion by Chathu07** [fail2ban/fail2ban#3321] [Q&A] [answered]
I have set up fail2ban to track successful login via the Ubuntu machine's `auth.log`. Once the user logged in via SSH, SFTP to my server, or WordPress sites, I configured fail2ban to track that IP address via `auth.log`.  This jail works as I want, but there is a problem with whitelisting. 

I want to whitelist a successful login user for 30 or 60 minutes for all active fail2ban jails. Is there any possibility of temporarily adding an IP address to the `ignore IP` list via **action.d**?  (whitelist IP for all jails)

Right now, I use the following action.d, but the problem is if fail2ban other filters trigger any jail, it instantly blocks that IP address. Because it only whitelists single jail, I have over 15 jails.  

How do I add an IP address to ignore the list on a temporary (auto expire) basis?

```
[f2b-whitelist]
enabled   = true
port      = http,https
filter    = f2b-whitelist
logpath   = /var/log/auth.log
blocktype    = RETURN
returntype   = DROP
action    = ignoreip
                pushover
findtime = 1800
maxretry  = 1
bantime   = 1800
```

```
[Definition]
actionstart =
actionstop  =
actioncheck = iptables -n -L <chain> | grep -q 'f2b-<name>[ \t]'
actionban   = fail2ban-client set <name> addignoreip <ip>
actionunban = fail2ban-client set <name> delignoreip <ip>

[Init]
name  = default
chain = INPUT
```

**Comment by sebres** [comment, marked as answer]
You have several possibilities to achieve that:
1. pass2allow: create a jail with custom action (or parameters) whitelisting some chain (before INPUT chain in the rules of net-filter) and bantime 30 or 60 minutes, so the bans in other jails would be bypassed (don't enter all `f2b-*` rules of INPUT chain). 
Example can be the jail `pass2allow-ftp` (just you'd need another set of ports or allports-action):
https://github.com/fail2ban/fail2ban/blob/3c22ae161cfe23a9fb00cabb8a7de891e3e8edaf/config/jail.conf#L911-L925
2. an approach with `ignoreip` action (similar your action), just it must get a cycle around `fail2ban-client set "$name" addignoreip <ip>`  by `$name` notifying all jails you need.
3. config-reload (untested): similar 2., but instead of notifying fail2ban directly, write some file containing following content and add it to includes of `jail.local`, so your action would extend that file and invoke `fail2ban-client reload` after the IP gets added/removed from that set...
```
[DEFAULT]
ignoreip = 192.0.2.1 192.0.2.2 ....
```

4. if you have some NS server (e. g. locally), you could create own FQDN (with small refresh/expiry TTLs) which you'd extend within your action and add this FQDN to `ignoreip` (as domain-name instead of IP).
5. [ignorecommand](https://manpages.debian.org/unstable/fail2ban/jail.conf.5.en.html#ignorecommand) - a command which will be invoked to check programmatically whether the IP must be ignored. So you can store your list wherever you want (e. g. in SQL-database) and don't need to notify fail2ban directly. You can also cache it to speedup lookup a bit, using [ignorecache](https://manpages.debian.org/unstable/fail2ban/jail.conf.5.en.html#ignorecache) parameter.
For example, see https://github.com/fail2ban/fail2ban/issues/2013#issuecomment-426940189 answering similar question.

I'd recommend to use method 5 (with `ignorecommand`) or 4 (with own NS server).
